src/frontend/js/utils/api/web-analytics/index.ts

```diff
@@ -0,0 +1,19 @@
+import { WebAnalyticsAPI, WebAnalyticsAPIBackend } from 'types/WebAnalytics';
+import context from 'utils/context';
+import GoogleAnalyticsApiInterface from './google_analytics';
+import GoogleTagManagerApiInterface from './google_tag_manager';
+import BaseWebAnalyticsApiInterface from './base';
+
+const WEB_ANALYTICS_PROVIDER = context.web_analytics_provider || '';
+
+const WebAnalyticsAPIHandler = (): WebAnalyticsAPI => {
+  switch (WEB_ANALYTICS_PROVIDER) {
+    case WebAnalyticsAPIBackend.GOOGLE_ANALYTICS:
+      return GoogleAnalyticsApiInterface();
+    case WebAnalyticsAPIBackend.GOOGLE_TAG_MANAGER:
+      return GoogleTagManagerApiInterface();
+  }
+  return BaseWebAnalyticsApiInterface();
```

Comment:
If no web analytics is defined within context we should return `undefined`
```suggestion
  return undefined;
```

Then when you want to use the `WebAnalyticsAPIHandler` you just have to use an optional chaining operator.

Reply:
✔